src/server/linux.rs

```diff
@@ -0,0 +1,268 @@
+    -> Result<VersionResult, anyhow::Error>
+{
+    let mut max_version = None::<&PackageInfo>;
+    for package in &haystack.packages {
+        if version_matches(package, ver) {
+            if let Some(max) = max_version {
+                if max.version < package.version {
+                    max_version = Some(package);
+                }
+            } else {
+                max_version = Some(package);
+            }
+        }
+    }
+    if let Some(target) = max_version {
+        let major = target.slot.as_ref().unwrap().clone();
+        Ok(VersionResult {
+            package_name:
+                if major.to_ref() >= Version("1-alpha3") {
+                    "edgedb-server".into()
+                } else {
+                    "edgedb".into()
+                },
+            major_version: major,
+            version: target.version.clone(),
+            revision: target.revision.clone(),
+        })
+    } else {
+        anyhow::bail!("Version {} not found", ver)
+    }
+}
+
+pub fn perform_install(operations: Vec<Operation>, linux: &Linux)
+    -> anyhow::Result<()>
+{
+    let mut ctx = operation::Context::new();
+    let has_privileged = operations.iter().any(|x| x.is_privileged());
+    if has_privileged && linux.get_user_id() != 0 {
+        println!("The following commands will be run with elevated \
+            privileges using sudo:");
+        for op in &operations {
+            if op.is_privileged() {
+                println!("    {}", op.format(true));
+            }
+        }
+        println!("Depending on system settings sudo may now ask \
+                  you for your password...");
+        match linux.get_sudo_path() {
+            Some(cmd) => ctx.set_elevation_cmd(cmd),
+            None => {
+                eprintln!("`sudo` command not found. \
+                           Cannot elevate acquire needed for \
+                           installation. Please run \
+                           `edgedb server install` as root user.");
+                exit(exit_codes::NO_SUDO);
+            }
+        }
+    }
+    for op in &operations {
+        op.perform(&ctx)?;
+    }
+    Ok(())
+}
+
+pub fn get_server_path(major_version: &Version<String>)
+    -> anyhow::Result<PathBuf>
+{
+    Ok(Path::new("/usr/lib")
+        .join(&format!("{}-linux-gnu", ARCH))
+        .join(&format!("edgedb-{}", major_version))
+        .join("bin/edgedb-server"))
+}
+
+pub fn systemd_unit(settings: &init::Settings)
+    -> anyhow::Result<String>
+{
+    Ok(format!(r###"
+[Unit]
+Description=EdgeDB Database Service, instance {instance_name:?}
+Documentation=https://edgedb.com/
+After=syslog.target
+After=network.target
+
+[Service]
+Type=notify
+
+Environment=EDGEDATA={directory}
+RuntimeDirectory=edgedb-{instance_name}
+
+ExecStart={server_path} -D ${{EDGEDATA}} --runstate-dir=%t/edgedb-{instance_name}
+ExecReload=/bin/kill -HUP ${{MAINPID}}
+KillMode=mixed
+KillSignal=SIGINT
+TimeoutSec=0
```

Comment:
Looks like this will be a problem on distros like CentOS 8 with [systemd >= 229](https://github.com/systemd/systemd/blob/a2031de849da52aa85b7e4326c0112ed7e5b5672/NEWS#L6440-L6444) if you meant to turn off the timeout here.

Reply:
> To maintain backwards compatibility, "0" continues turn off previously existing timeout settings.

I think this is previously existing timeout, so fine. I think we need to support older systemd for now, but I did no research (i.e. we only need to support as old as systemd on a system we have packages for).